### PR TITLE
#6819: Enhance SLD parser for IconSymbolizer

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
This PR enhances the identification and parsing of format for image with no extension/format for the geostyler-sld-parser to use to generate `Format` node in `ExternalGraphic` for IconSymbolizer

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix

## Issue

**What is the current behavior?**
#6819 

**What is the new behavior?**
- When image input is missing format, then user is presented with format selector to choose from
- When style editor is of CSS, then format selector is not shown, as the css parses the url correctly and updates the style regardless of whether format present or not
- When image input has extension/format or when the image source is not valid or when error, the format selector is not shown

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/translations/data.en-US.json [hunk 1]
```diff
@@ -2243,6 +2244,7 @@
             "imageSrcLoadError": "The provided image source could not be loaded. Try with a different url or ensure the source is reachable by MapStore",
             "imageSrcInvalidBase64": "This is not a valid base64 image. Some style renderer does not support this type of source, you could use an external url source instead",
             "imageSrcNotSupportedBase64Image": "Some style renderer does not support base64 images even if the current source is valid",
+            "imageFormatEmpty": "Parser unable to identify format of the image even when the source is valid. Please select the format of the image provided from the dropdown",
```
Comment (allyoucanmap) [inline web/client/translations/data.en-US.json:2247]: I propose to change this message with something less technical:

`The image format cannot be identified. Please select the image format from the available ones in the 'Format' dropdown.`

File: web/client/translations/data.it-IT.json [hunk 2]
```diff
@@ -2233,6 +2234,7 @@
             "imageSrcLoadError": "Impossibile caricare l'origine dell'immagine fornita. Prova con un URL diverso o assicurati che la fonte sia raggiungibile da MapStore",
             "imageSrcInvalidBase64": "Questa non è un'immagine base64 valida. Alcuni renderer di stili non supportano questo tipo di risorsa, potresti invece utilizzare una immagine con URL esterno",
             "imageSrcNotSupportedBase64Image": "Alcuni renderer di stili non supportano le immagini base64 anche se la risorsa è valida",
+            "imageFormatEmpty": "Il parser non è in grado di identificare il formato dell'immagine anche quando l'origine è valida. Seleziona il formato dell'immagine fornita dal menu a discesa",
```
Comment (allyoucanmap) [inline web/client/translations/data.it-IT.json:2237]: `Non è possibile identificare il formato dell'immagine. Seleziona il formato dell'immagine fra quelli disponibili nel campo 'Formato'.`

File: web/client/components/styleeditor/config/blocks.js [hunk 3]
```diff
@@ -68,6 +70,16 @@ const getBlocks = (/* config = {} */) => {
                     label: 'styleeditor.image',
                     key: 'image'
                 }),
+                format: property.select({
+                    label: 'styleeditor.format',
+                    key: 'format',
+                    isVisible: (value, { image } = {}, format) => {
+                        return format !== 'css'
+                        && !isObject(image)
```
Comment (allyoucanmap) [inline web/client/components/styleeditor/config/blocks.js:78]: When do we get an image object here?

File: web/client/components/styleeditor/Fields.jsx [hunk 4]
```diff
@@ -225,9 +229,13 @@ export const fields = {
 
         const [newOptions, setNewOptions] = useState(initOptions(options));
         useEffect(() => {
-            !isEqual(options, newOptions) && setNewOptions(initOptions(options));
+            !multi && !isEqual(options, newOptions) && setNewOptions(initOptions(options));
```
Comment (allyoucanmap) [inline web/client/components/styleeditor/Fields.jsx:232]: is this the missing control that was causing the application freeze?

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/components/styleeditor/Fields.jsx (hunk 4 above)
Comment (dsuren1) [reply web/client/components/styleeditor/Fields.jsx:232]: Yes, that's right

**Review by allyoucanmap — APPROVED** [review]

**Comment by allyoucanmap** [comment]
@ElenaGallo @tdipisa this fix needs to be tested for all Symbolizers where it's possible to use the image icon (Icon, Line with Patthern and Polygon with Pattern). The format field shows up when it's not possible to detect the image format and it's needed only for SLD right now but could you please test also CSS part to verify the correct render of images?

After the test we need to backport this fix to this branch https://github.com/geosolutions-it/MapStore2/tree/geonode-mapstore-client-2.1.x

**Review by allyoucanmap — APPROVED** [review]

**Comment by tdipisa** [comment]
@ElenaGallo this PR should fix also #6899, so please thest this issue too.

**Comment by ElenaGallo** [comment]
@tdipisa test done, it is now possible to backport this PR.

**Pull request merged**